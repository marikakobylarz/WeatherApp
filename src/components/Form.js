import React from 'react';

const Form = (props) => {
    return (
        <form className="form" onSubmit={props.submit}>
            <input tyle="text"
                className="form__input"
                value={props.value}
                placeholder="Wpisz miasto"
                onChange={props.change}></input>
            <button className="form__button" onSubmit={props.submit}>Szukaj</button>
        </form>
    );
}
export default Form;