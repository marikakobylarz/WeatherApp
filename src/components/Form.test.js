import React from "react";
import Form from "./Form.js";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

describe('<Form/>', () => {
    it("renders correctly", () => {
        const wrapper = shallow(
            <Form onSubmit={() => { }} value="london" onChange={() => { }} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("Has an input field", () => {
        const wrapper = shallow(<Form />);
        expect(wrapper.find(".form__input").length).toEqual(1);
    });
    it("Has an search button", () => {
        const wrapper = shallow(<Form />);
        expect(wrapper.find(".form__button").length).toEqual(1);
    });
});


