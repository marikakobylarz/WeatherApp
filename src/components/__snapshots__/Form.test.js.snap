// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders correctly 1`] = `
<form
  className="form"
>
  <input
    className="form__input"
    placeholder="Wpisz miasto"
    type="text"
    value="london"
  />
  <button
    className="form__button"
    style={
      Object {
        "backgroundImage": "url(glass.png)",
      }
    }
  />
</form>
`;
